include/device.h

```diff
@@ -9,6 +9,7 @@
 #define ZEPHYR_INCLUDE_DEVICE_H_
 
 #include <kernel.h>
+#include <dts_device_extern.h>
```

Comment:
Why not instead including it as the other generated_dts_*.h files?

Reply:
Not sure I follow.

Reply:
including it in generated_dts_board.h
I find it awkward to have this new generated header in a public header file such as device.h

Reply:
There's an issue in that generated_dts_board.h is just defines, putting `externs` in there causes issues with the linker scripts as they include and I think utilize defines from generated_dts_board.h

Reply:
can't we `#ifndef _LINKER_SCRIPT` our way around that?

Reply:
Or advance to putting them all into a linker section in ZDID order, and indexing into the resulting array with ZDID.  A benefit of that is that the opaque handle could be a `u16_t`, we could retain `device_get_binding(ID)` with a change in argument signature (or a parallel function with different name), and we don't have to worry as much about people trying to use uninitialized devices just because they got the pointer to one somehow.  Then there's just:
```
extern struct device *__devices[];
#define device_get_dt_binding(BLAH) __devices[MAGIC_GET_ZDID(BLAH)]
```
somewhere.

(I assume one ordinal reserved for "no-such-device", and no Zephyr image would have more than 65534 devices.  254 seems too small.)

Reply:
> can't we `#ifndef _LINKER_SCRIPT` our way around that?

possibly, didn't know this ifdef existed.

Reply:
> possibly, didn't know this ifdef existed.

The name is just a suggestion, I'm saying we can make sure a macro is defined only when running cpp over the ld scripts and use that here